src/Services/ActionManager.vala

```diff
@@ -452,6 +456,45 @@ public class Akira.Services.ActionManager : Object {
         dialog.present ();
     }
 
+    private void action_pick_color () {
+        if (window.main_window.main_canvas.canvas.selected_bound_manager.selected_items.length() == 0) {
+            return;
+        }
+        bool is_holding_shift = false;
+        var color_picker = new Akira.Utils.ColorPicker ();
+        color_picker.show_all();
+        color_picker.key_press_event.connect((e) => {
+            if (e.keyval == Gdk.Key.Shift_R) {
+                is_holding_shift = true;
+            }
+            return true;
+        });
+        color_picker.key_release_event.connect((e) => {
+            if (e.keyval == Gdk.Key.Shift_R) {
+                is_holding_shift = false;
+            }
+            return true;
+        });
+        color_picker.cancelled.connect(() => {
+            color_picker.close();
+        });
+        color_picker.picked.connect(color => {
+            if (!is_holding_shift) {
+                // TODO change current selected shape fill color
+                for (int i = 0; i <= window.main_window.main_canvas.canvas.selected_bound_manager.selected_items.length (); i ++) {
+                    var item = window.main_window.main_canvas.canvas.selected_bound_manager.selected_items.nth_data(i);
+                    if(item != null) {
+                        item.color = color;
+                    }
+                }
+            } else {
+                // TODO change current selected shape border color
+                // window.main_window.main_canvas.canvas.selected_bound_manager.delete_selection ();
+            }
+            color_picker.close();
+        });
+    }
+
```

Comment:
I updated the method with this: 
```suggestion
    private void action_pick_color () {
        weak Akira.Lib.Canvas canvas = window.main_window.main_canvas.canvas;
        // Interrupt if no item is selected.
        if (canvas.selected_bound_manager.selected_items.length () == 0) {
            return;
        }

        foreach (var item in canvas.selected_bound_manager.selected_items) {
            // Hide the ghost bound manager.
            item.bounds_manager.hide ();
        }

        bool is_holding_shift = false;
        var color_picker = new Akira.Utils.ColorPicker ();
        color_picker.show_all ();

        color_picker.key_press_event.connect (e => {
            is_holding_shift = e.keyval == Gdk.Key.Shift_R;
            return true;
        });

        color_picker.key_release_event.connect (e => {
            is_holding_shift = e.keyval == Gdk.Key.Shift_R;
            return true;
        });

        color_picker.cancelled.connect (() => {
            color_picker.close ();
        });

        color_picker.picked.connect (color => {
            foreach (var item in canvas.selected_bound_manager.selected_items) {
                if (is_holding_shift) {
                    item.border_color_string = Utils.Color.rgba_to_string (color);
                } else {
                    item.color_string = Utils.Color.rgba_to_string (color);
                }
                item.load_colors ();
            }
            color_picker.close ();
        });
    }
```

It properly updates the fill color, but it doesn't update the fill or border widget in the options panel as those models are binded the other way around (model -> item), instead here we're doing the opposite (item -> model).

I think I'm fine with merging this as it is, as I'm not sure we can directly edit the fill item or the fill item model.
Maybe a restructure of that section is necessary, or maybe not, I don't know :sweat_smile:

Reply:
Oh, this suggestion won't work without a tiny update to the `Color.vala`

```
public static string rgba_to_hex (string rgba_string) {
        var rgba = Gdk.RGBA ();
        rgba.parse (rgba_string);

        return rgba_to_string (rgba);
    }

    public static string rgba_to_string (Gdk.RGBA rgba) {
        return "#%02x%02x%02x".printf (
            (int) (rgba.red * 255),
            (int) (rgba.green * 255),
            (int) (rgba.blue * 255)
        );
    }
```